### PR TITLE
Fix test_unaligned_bit_chunk_iterator

### LINKED ISSUE #1294: Test failure: bit_chunk_iterator
**Describe the bug**
The `test_unaligned_bit_chunk_iterator()` test fails to run on aarch64 because of an assertion `assert!(ALIGNMENT > 64)`.  There are several platforms that support 64-bit or lower alignment, and this assertion will also fail on them.

**To Reproduce**

Run arrow tests with arm, aarch64 or other platform with a lower alignment than the assertion

**Expected behavior**

Not sure of what the implication of > 64 alignment is, as I haven't been following the codebase

**Additional context**

Here's the alignment for aarch64 https://github.com/apache/arrow-rs/blob/master/arrow/src/alloc/alignment.rs#L119. CC @tustvold from #1228

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #1294

# Rationale for this change
 
Test should pass

# What changes are included in this PR?

Fix test

# Are there any user-facing changes?

No, this only changes a test

### CODE REVIEW

File: arrow/src/util/bit_chunk_iterator.rs [hunk 1]
```diff
@@ -466,8 +466,8 @@ mod tests {
     #[test]
     #[allow(clippy::assertions_on_constants)]
     fn test_unaligned_bit_chunk_iterator() {
-        // This test exploits the fact Buffer is at least 64-byte aligned
-        assert!(ALIGNMENT > 64);
+        // This test exploits the fact Buffer is at least aligned to a u64
+        assert!(ALIGNMENT >= std::mem::align_of::<u64>());
```
Comment (tustvold) [inline arrow/src/util/bit_chunk_iterator.rs:470]: The major change here is to `>=` which was just a :facepalm: 

There are platforms where ALIGNMENT is only 4 bytes, but if this assertion doesn't hold we have a broader issue here - as we assume that the data allocated by `Buffer` can be `align_to::<u64>` without remainder.

**Comment by tustvold** [comment]
I have tested this on an aarch64 M1 Mac and it now passes 😀

File: arrow/src/util/bit_chunk_iterator.rs [hunk 2]
```diff
@@ -466,9 +466,6 @@ mod tests {
     #[test]
     #[allow(clippy::assertions_on_constants)]
     fn test_unaligned_bit_chunk_iterator() {
-        // This test exploits the fact Buffer is at least 64-byte aligned
-        assert!(ALIGNMENT > 64);
```
Comment (tustvold) [inline arrow/src/util/bit_chunk_iterator.rs:470]: The derp here is strong. The test requires that `Buffer is 64 **bit** aligned, not 64 **byte** aligned :facepalm:

As an added bonus, even if it did care, this should be `>=`

File: arrow/src/util/bit_chunk_iterator.rs [hunk 3]
```diff
@@ -520,6 +516,13 @@ mod tests {
         assert_eq!(unaligned.suffix(), None);
 
         let buffer = Buffer::from(&[0xFF; 14]);
+
+        // Verify buffer alignment
+        let (prefix, aligned, suffix) = unsafe { buffer.as_slice().align_to::<u64>() };
```
Comment (tustvold) [inline arrow/src/util/bit_chunk_iterator.rs:521]: Technically `align_to` is not guaranteed to return the maximal `aligned` slice. In practice we rely on this in so many places, this is what the upstream code currently does, and any other behaviour would be extremely contrived.

**Review by jhorstmann — APPROVED** [review]

**Review by nevi-me — APPROVED** [review]

**Comment by alamb** [comment]
@nevi-me  would you like me to spin up a new RC with this change?

**Comment by nevi-me** [comment]
> @nevi-me would you like me to spin up a new RC with this change?
> 
> Edit: or are you comfortable with releasing 9.0.3 given that this is only a test issue?

I'm away from my laptop, I didn't see if the failure prevented other crates from running, or if they ran. I'll confirm this evening.

I don't think cutting another RC would be necessary. Thanks @alamb

**Pull request merged**